### ISSUE TITLE
ST++: Make Self-training Work Better for Semi-supervised Semantic Segmentation

### ISSUE THREAD

**Issue by kabupen** [kabupen/papers#4]
# 

- 2021
- [ST++: Make Self-training Work Better for Semi-supervised Semantic Segmentation](https://arxiv.org/pdf/2106.05095.pdf)

## 経緯

- Deep Anything を読んでいて気になったので

# Abstract

- セマンティックセグメンテーションに対する半教師あり自己学習（self-training）についてのベースラインを作成した。
- 強い augmentation を掛けることで種々の問題に対応した
- selective re-trainig を提案した

# Introduction

semi-self trainig（半教師あり学習）とは、ラベル付きデータとラベルなしデータを用いて学習する手法で、ラベル付きデータで一旦学習してラベルなしデータに擬似ラベルを振り、それを学習データとして用いてさらに..というサイクルを回す。semi-self trainig の実装方法はいくつかあるらしく、その中の手法の一つが self-trainig である。

# Related Works

## Semi-supervised learning（SSL）

半教師あり学習は、少量のアノテーション済みデータとラベルなしデータを学習データとして用いる手法。自分的に抑えておかないといけないのは、疑似ラベルを振るだけではないということ。

- consistency regularization
    - 様々な外乱に対してモデルが robust になるように学習する手法
- entropy minimization
    - ???  

という手法。その他 [55] で示されているのは

- consistency methods [[FixMatch](https://arxiv.org/abs/2001.07685)]
- pseudo-labelling methods
- representation learning

### FixMatch

FixMatch では consistency regularization と pseudo-labeling を合わせた以下の方法で学習している。弱く agumentation をかけたものと、強い augmentation をかけたもののクラス分類結果が同じになるように学習させる。

![image](https://github.com/kabupen/papers/assets/19812756/7bf87269-db6f-4b07-b8bd-fe70bf943b1e)

- $H(p, q)$ はクロスエントロピーで、出てきた確率分布が近くなるように学習を進めている

## self-training 

疑似ラベルを用いた自己学習は [33] に発端する classical method である。近年より注目を浴びているらしい。
先行研究の [55] では

> the success of supervised semantic segmentation typically relies on a large amount of labeled data, which is time-consuming and costly to obtain

とあるように、（一般的だが）アノテーションコストをかければそれなりのモデルができることはすでに分かっていると主張している。

## Self-supervised learning（SSL）

masked language models で用いられている「入力文をマスクしてそれを予測する形で学習を進める」手法のように、（完全に）教師モデルを必要としない学習手法のこと。アノテーションコストが不要である。

# Method

## Problem definition

ラベル付きデータ $D^l$ とラベルなしデータ $D^u$ を用いて semi-supervised learning を行う。

$$
L = L^s + \lambda L^u
$$

## Plainest Self-trainig Scheme

- 教師モデルをラベル付きデータで学習
- 疑似ラベルを生成
- 再学習

unsupervised loss は

$$
L_{\text{plain}}^{u} = H(T(x), S(A^{u}(x)))
$$

で表現できる。$T(x)$ は教師モデルの出力、$S(x)$ は生徒モデルの出力、$A(x)$ は augmentation をかけていることを表現している。

## ST: Inject SDA on Unlabeled Images

教師モデルをラベル付きデータで学習 --> 疑似ラベルを生成 --> 再学習 というループを回す self-trainig は長らく批判されているもの。

- 疑似ラベルの精度の悪さが積み重なって student モデルの精度を劣化させてしまう
- student モデルは teacher モデルの疑似ラベルを真の正解だと思って学習を進めるのだが、そもそも $S$ と $T$ は同じ構造で同じ初期値だったりすると有用な extra information を学習できない

ノイズの多い（ミスラベルの多い）データに対しての overfitting とcoupling issue に対応するにはどうすればいいか？ --> **強い augmentation をデータに付与する**

補足：ここでの coupling issue とは教師と生徒モデルの予測が密接に関連しすぎている状況を指し、生徒モデルが教師モデルの誤ったラベルを学習してしまい、両モデルの予測が過度に類似してしまうことかな？

### Disucussion

[12, 25, 66] の online method と比較して、本論文では疑似ラベルは1回のみ振る。

## ST++: Select and Prioritize Reliable Images

ただし疑似ラベルは各サンプルのアノテーションしづらさみたいなものを考慮せずにつけられているものなので、その信頼性などは考慮されていない。そこで ST++ では selective re-trainig scheme を導入している。

予測（疑似ラベル）の信頼性、誤差をいかに評価するか？

教師モデル学習時の各エポックのチェックポイントを用いて、stability score を算出する。

![image](https://github.com/kabupen/papers/assets/19812756/4203174d-3050-4214-ad1e-467f650caac3)

- ラベル付きデータで教師モデル $T$ を学習する
- そのチェックポイントを使って、unlabled データをアノテーションする
- 各 $T$ モデルの予測が近しいものは精度が一定している = 信頼度が高いと判定して、生徒モデルの学習に使用する
    - こうすることで「〇〇より高いものを信頼度が高い」 のようにしきい値を設定しなくてすむ
- 残っていたデータセットを生徒モデルでアノテーションして、再学習する

# メモ

- 33 Pseudo-label: The simple and efficient semi-supervised learning method for deep neural networks
- 55 [A simple baseline for semi-supervised semantic segmentation with strong data augmentation.](https://arxiv.org/abs/2104.07256)
- 12 [Semi-supervised semantic segmentation with cross pseudo supervision.]()
- 25, Semi-supervised semantic segmentation via adaptive equalization learning.
- 66, Pseudoseg: Designing pseudo labels for semantic segmentation.
